### PR TITLE
Add GPU tracking to Jobs realm

### PR DESCRIPTION
**Pull request by jtpalmer**
## Description

Adds GPU statistics and a group by to the jobs realm.  Only includes support for Slurm at this time.

Replaces #1235 

## Motivation and Context

https://app.asana.com/0/15230011293693/1145503330514643

## Tests performed

Added additional tests and regression test data.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project as found in the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: classes/OpenXdmod/Migration/Version851To900/DatabaseMigration.php [hunk 1]
```diff
@@ -29,5 +35,63 @@ public function execute()
             $this->logger->warning('Failed to build filter list: '  . $e->getMessage());
             $this->logger->warning('You may need to run xdmod-build-filter-lists manually');
         }
+
+        $console = Console::factory();
+        $console->displayMessage(<<<"EOT"
+This version of Open XDMoD has support for GPU metrics in the jobs realm.  If
+you have shredded Slurm job records in the past it is possible to extract the
```
Comment (jpwhite4) [inline classes/OpenXdmod/Migration/Version851To900/DatabaseMigration.php]: What version of XDMoD was the reqgres data added? Or has the capability been there effectively forever?

Comment (jtpalmer) [reply classes/OpenXdmod/Migration/Version851To900/DatabaseMigration.php]: It was added in 6.5 (released 2017-01-10).

Comment (jpwhite4) [reply classes/OpenXdmod/Migration/Version851To900/DatabaseMigration.php]: Perhaps just a note in the markdown docs about this. I doubt many people have pre 6.5 data in their database.

Comment (jtpalmer) [reply classes/OpenXdmod/Migration/Version851To900/DatabaseMigration.php]: I added a note to the upgrade documentation.

**Pull request merged**